### PR TITLE
User delete workflow fixes for #1365

### PR DESCRIPTION
**Pull request by micahmills**
adds a function on the 'delete_user' hook to delete the 'correspons_t…o_user' meta key and value.

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -485,3 +485,16 @@ function dt_theme_admin_notice_required_php_version() {
     </div>
     <?php
 }
+
+add_action( 'delete_user', 'dt_delete_user_contact_meta' );
+function dt_delete_user_contact_meta( $user_id ) {
+    global $wpdb;
+    // phpcs:disable
+    // WordPress.WP.PreparedSQL.NotPrepared
+    $wpdb->get_results( "DELETE FROM $wpdb->postmeta pm
```
Comment (corsacca) [inline functions.php:494]: @micahmills this one should work fine using $wpdb->prepare() so with don't have to disable phpcs

**Comment by corsacca** [comment]
Thanks @micahmills !
We can put this function in dt-users/users.php

**Comment by corsacca** [comment]
If a user is deleted from the network side, this might need to run on each subsite.

**Comment by micahmills** [comment]
According to the Wordpress codex (https://developer.wordpress.org/reference/hooks/delete_user/) doesn't run on all multisite situations when a user is deleted, but if we change to the 'deleted_user' hook it should run. I will play with it and see what it does.

**Comment by micahmills** [comment]
Something to note here is that the record type continues to be a 'user.' We could revert it to another user type in this function, but there are possible security concerns with changing it to an access record when it should be a private record.

**Comment by corsacca** [comment]
> According to the Wordpress codex (https://developer.wordpress.org/reference/hooks/delete_user/) doesn't run on all multisite situations when a user is deleted, but if we change to the 'deleted_user' hook it should run. I will play with it and see what it does.

`deleted_user` Fires immediately after a user is deleted from the database.

i wonder if the `remove_user_from_blog` hook would work better. It would be called for each site

**Comment by corsacca** [comment]
> Something to note here is that the record type continues to be a 'user.' We could revert it to another user type in this function, but there are possible security concerns with changing it to an access record when it should be a private record.

Yeah. i don't know how best to solve this one. I think we keep it as a user record and change the status to archived.

**Comment by prykon** [comment]
I'm able to delete a user and then re-create it for the same contact, but there's a new bug:
The display name gets ignored. I set a display name for the user but it still shows the contact name. This worked fine with the previous code.

**Comment by micahmills** [comment]
Where is the display name not displaying?

**Comment by prykon** [comment]
In the contacts page and in the contact details page.
It does, however, appear in the wp-admin/users.php and in user-management/users/

**Comment by prykon** [comment]
I re cloned the branch and it works fine now. Must have been an old file that was still in there somewhere.

**Comment by corsacca** [comment]
thanks @micahmills  and @prykon !

**Pull request merged**